support awsic plugin `--assume-role-arn` flag

flyinghermit opened the pull request:
Proto and type changes already merged with https://github.com/gravitational/teleport/pull/53024

Plugin runtime credential handled in https://github.com/gravitational/teleport.e/pull/6170

changelog: The `tctl` based AWS IAM Identity Center integration installer now supports `--assume-role-arn` flag that instructs integration client to assume AWS role instead of directly using the system credential.

flyinghermit committed:
support awsic plugin assume-role-arn flag

r0mant commented on tool/tctl/common/plugin/awsic.go:
```diff
@@ -71,6 +72,22 @@ func (a *awsICArgs) validate(ctx context.Context, log *slog.Logger) error {
 
+func (a *awsICArgs) validateSystemCredentialInput() error {
+	if !a.useSystemCredentials {
+		return trace.BadParameter("--useSystemCredentials must be set")
```
I think the flag name is incorrect?
```suggestion
		return trace.BadParameter("--use-system-credentials must be set")
```

r0mant commented on tool/tctl/common/plugin/awsic.go:
```diff
@@ -71,6 +72,22 @@ func (a *awsICArgs) validate(ctx context.Context, log *slog.Logger) error {
+	}
+
+	if a.assumeRoleARN == "" {
+		return trace.BadParameter("--assume-role-arn must be set when --useSystemCredentials is configured")
```
```suggestion
		return trace.BadParameter("--assume-role-arn must be set when --use-system-credentials is configured")
```

r0mant reviewed (approved):
lgtm with 2 typos but also please make sure to update the documentation

kopiczko commented on tool/tctl/common/plugin/awsic.go:
```diff
@@ -71,6 +72,22 @@ func (a *awsICArgs) validate(ctx context.Context, log *slog.Logger) error {
 }
 
+func (a *awsICArgs) validateSystemCredentialInput() error {
+	if !a.useSystemCredentials {
+		return trace.BadParameter("--useSystemCredentials must be set")
+	}
```
Does it mean this flag has to always be set?

Possibly the error message should be:

--use-system-credentials must be set, currently only AWS Local system credentials are supported

flyinghermit replied:
Done. It now says "--use-system-credentials must be set. The tctl-based AWS IAM Identity Center plugin installation only supports AWS local system credentials"

flyinghermit committed:
Merge branch 'master' into sshah/ic-assume-role

flyinghermit committed:
review comments:
- fix command typo useSystemCredentials -> use-system-credentials
- explain why use-system-credentials is required
